runtime/interpreter/encode.go

```diff
@@ -699,6 +701,25 @@ func (s SomeStorable) Encode(e *atree.Encoder) error {
 	return s.Storable.Encode(e)
 }
 
+// EncodeAsElement encodes SomeStorable as
+//
+//	cbor.Tag{
+//			Number: CBORTagSomeValue,
+//			Content: Value(v.Value),
+//	}
+func (s SomeStorable) EncodeAsElement(e *atree.Encoder, inlinedExtraData atree.InlinedExtraData) error {
```

Comment:
Why is this duplicate of `SomeStorable.Encode` needed?

Reply:
> Why is this duplicate of SomeStorable.Encode needed?

`SomeStorable.EncodeAsElement` and `SomeStorable.Encode` have a different API and are used differently.

```Go
Encode(*Encoder) error

EncodeAsElement(*Encoder, InlinedExtraData) error
```

`EncodeAsElement` is function in `atree.ContainerStorable` interface.  `ContainerStorable` is implemented by storable containing other storables, such as `atree.ArrayDataSlab` and `atree.MapDataSlab`.  Those storables and their child storables may need to be encoded differently when they are an element of parent array/map.  

`SomeStorable` implements `atree.ContainerStorable` as well because it contains another storable and that storable can be `atree.ArrayDataSlab` that needs to be encoded differently.

On the other hand, I don't know of a use case for `SomeStorable.Encode()`.  If `SomeStorable.Encode()` returns `UnreachableError`, no cadence tests fail as result of it. I left it as is for now, and Cadence team can decide which approach makes more sense.

Reply:
Thank you for the great explanation. Could you please add it to the function as a docstring on the atree side? See https://github.com/onflow/atree/pull/352/files#r1366193018